docs/cody/core-concepts/prompting-guide.mdx

```diff
@@ -0,0 +1,199 @@
+```
+
+### Provide Specific Context
+
+Providing detailed context is the most effective way to enhance Cody's assistance. Approach this as if you are explaining the situation to a new engineer.
+
+You should:
+
+-   Reference important files and symbols
+-   Provide examples from other similar functions
+
+To provide specific context from your codebase, you can use the `@-` mention feature to provide syntax. You can mention specific files, sections of code, and symbols to help Cody understand your code.
+
+Here is an example of bringing in specific lines of code. The shortcut to do this is to highlight code and press `Opt+L` on Mac or `Ctrl+L` on Windows.
+
+IMAGE HERE: cody-mention-snippet.jpg
+
+Here is an example of `@-` mentioning a specific file:
+
+IMAGE HERE: cody-mention-file.jpg
+
+### Provide Examples and Test Cases
+
+Include examples and test cases when applicable to clarify your expectations. Demonstrate edge cases or handling of special conditions to guide Cody in generating robust code.
+
+❌ Bad example:
+
+```
+I need to validate email addresses in JavaScript
+```
+
+✅ Good example:
+
+```
+Create a function to validate email addresses in JavaScript. Return true for valid email addresses and false for invalid ones. Here are some example test cases:
+
+Valid:
+- "john@example.com"
+- "jane.doe@example.co.uk"
+
+Invalid:
+- "john@example"
+- "jane.doe@example."
+- "invalid.email"
+
+Please write the function
+```
+
+### Iterate and Refine
+
+Start with a general prompt and incrementally add more details based on Cody's responses. Take advantage of the fact that you can chat with Cody. Bring the back and forth conversation, especially if you didn't like Cody's first response. Review the generated code or suggestions, provide feedback, and refine your prompts to get the desired results.
+
+Initial prompt:
+
+```
+I need to calculate the total price of an order. Help me write the function
+```
+
+Refined prompt:
+
+```
+Thanks. I forgot to mention:
+
+- The function should take an array of order items as input
+- We need to apply a 10% discount if the total price exceeds $100
+- Final total should be rounded to two decimal places
+
+Please update the function
+```
+
+### Leverage Cody's Strengths
+
+Utilize Cody's capabilities for generating boilerplate code, common patterns, and repetitive tasks. Prompt it to assist with unit tests, documentation, and error handling to save time and ensure code quality.
+
+✅ Good example:
+
+```
+Help me write tests for the `calculateAverageRating` function. Here are the test cases I have in mind:
+
+- Empty ratings array should return 0
+- Array with one rating should return that rating
+- Array with multiple ratings should calculate the average correctly
+
+Make sure the tests cover any edge cases or potential issues. Thanks!
+```
+
+You can also use the **Generate Tests** command to generate tests for your code.
+
+## Conclusion
```

Comment:
```suggestion
## Conclusion
```
I'd suggest let's remove this conclusion part since it does not align with the docs content type. :)